### PR TITLE
CB-14421 CB-15971 targeted upscale fixes, improvements 

### PR DESCRIPTION
**Pull request by horadla23**

### CODE REVIEW

**Comment by oleewere** [comment]
for me highstate does not work after an upscale.

hosts pillar before upscale:
```bash
oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.13:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker1
        instance_id:
            i-012af315c5e208cd7
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.169:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-compute0
        instance_id:
            i-030360a2a40fbfafb
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.57:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker2
        instance_id:
            i-0cbf93650da2c44cd
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.199:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker0
        instance_id:
            i-09cbd57beaa11812f
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.13:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker1
        instance_id:
            i-012af315c5e208cd7
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.169:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-compute0
        instance_id:
            i-030360a2a40fbfafb
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.57:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker2
        instance_id:
            i-0cbf93650da2c44cd
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.199:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker0
        instance_id:
            i-09cbd57beaa11812f
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.13:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker1
        instance_id:
            i-012af315c5e208cd7
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.169:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-compute0
        instance_id:
            i-030360a2a40fbfafb
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.57:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker2
        instance_id:
            i-0cbf93650da2c44cd
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.199:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker0
        instance_id:
            i-09cbd57beaa11812f
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.13:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker1
        instance_id:
            i-012af315c5e208cd7
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.169:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-compute0
        instance_id:
            i-030360a2a40fbfafb
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.57:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker2
        instance_id:
            i-0cbf93650da2c44cd
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.199:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker0
        instance_id:
            i-09cbd57beaa11812f
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.13:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker1
        instance_id:
            i-012af315c5e208cd7
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.169:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-compute0
        instance_id:
            i-030360a2a40fbfafb
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.26.57:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker2
        instance_id:
            i-0cbf93650da2c44cd
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.199:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker0
        instance_id:
            i-09cbd57beaa11812f
        instance_type:
            m5.2xlarge
        public_address:
            False
```

hosts pillar after upscale (2 workers + gateway on all nodes);

```bash
oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.25.99:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker4
        instance_id:
            i-008f56f3de9b2054e
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.203:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker3
        instance_id:
            i-070c7955c1bfa4d7c
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.25.99:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker4
        instance_id:
            i-008f56f3de9b2054e
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.203:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker3
        instance_id:
            i-070c7955c1bfa4d7c
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.25.99:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker4
        instance_id:
            i-008f56f3de9b2054e
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.203:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker3
        instance_id:
            i-070c7955c1bfa4d7c
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.25.99:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker4
        instance_id:
            i-008f56f3de9b2054e
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.203:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker3
        instance_id:
            i-070c7955c1bfa4d7c
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.25.99:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker4
        instance_id:
            i-008f56f3de9b2054e
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.203:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker3
        instance_id:
            i-070c7955c1bfa4d7c
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.25.99:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker4
        instance_id:
            i-008f56f3de9b2054e
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.203:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker3
        instance_id:
            i-070c7955c1bfa4d7c
        instance_type:
            m5.2xlarge
        public_address:
            False
oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    ----------
    10.112.24.170:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-master0
        instance_id:
            i-0e2a9fd7720e03244
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.25.99:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker4
        instance_id:
            i-008f56f3de9b2054e
        instance_type:
            m5.2xlarge
        public_address:
            False
    10.112.27.203:
        ----------
        custom_domain:
            True
        domain:
            oszabo-l.xcu2-8y8x.wl.cloudera.site
        fqdn:
            oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site
        hostname:
            oszabo-de-worker3
        instance_id:
            i-070c7955c1bfa4d7c
        instance_type:
            m5.2xlarge
        public_address:
            False
```

grains only exists for the 3 related cluster (i guess that one is fine):

```bash
[root@oszabo-de-master0 cloudbreak]# /opt/salt_3000.8/bin/salt '*' grains.get instance_type
/opt/salt_3000.8/lib/python3.6/site-packages/salt/utils/jinja.py:656: DeprecationWarning: 'contextfunction' is renamed to 'pass_context', the old name will be removed in Jinja 3.1.
  @jinja2.contextfunction
oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    - m5.2xlarge
oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    - m5.2xlarge
oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    - m5.2xlarge
oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site:
oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site:
[root@oszabo-de-master0 cloudbreak]# /opt/salt_3000.8/bin/salt '*' grains.get instance_id
/opt/salt_3000.8/lib/python3.6/site-packages/salt/utils/jinja.py:656: DeprecationWarning: 'contextfunction' is renamed to 'pass_context', the old name will be removed in Jinja 3.1.
  @jinja2.contextfunction
oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
oszabo-de-worker4.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    - i-008f56f3de9b2054e
oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site:
oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site:
oszabo-de-worker3.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    - i-070c7955c1bfa4d7c
oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
oszabo-de-master0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    - i-0e2a9fd7720e03244
```

highstate errors:

```bash
oszabo-de-worker0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    Data failed to compile:
----------
    Rendering SLS 'base:unbound' failed: Jinja variable 'dict object' has no attribute '10.112.27.199'
/var/cache/salt/minion/files/base/nodes/settings.sls(4):
---
# https://tedops.github.io/how-to-find-default-active-ethernet-interface.html
{% set private_address = salt['network.interface_ip'](salt['network.default_route']('inet')[0]['interface']) %}
{% set host_details = salt['pillar.get']('hosts')[private_address] %}
{% set has_public_address = host_details['public_address'] %}    <======================
{% set instance_id = host_details['instance_id'] %}
{% set instance_type = host_details['instance_type'] %}

{% set host = {} %}
{% do host.update({
[...]
---
----------
    Rendering SLS 'base:metering' failed: Jinja variable 'dict object' has no attribute '10.112.27.199'
/var/cache/salt/minion/files/base/nodes/settings.sls(4):
---
# https://tedops.github.io/how-to-find-default-active-ethernet-interface.html
{% set private_address = salt['network.interface_ip'](salt['network.default_route']('inet')[0]['interface']) %}
{% set host_details = salt['pillar.get']('hosts')[private_address] %}
{% set has_public_address = host_details['public_address'] %}    <======================
{% set instance_id = host_details['instance_id'] %}
{% set instance_type = host_details['instance_type'] %}

{% set host = {} %}
{% do host.update({
[...]
---
oszabo-de-worker1.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    Data failed to compile:
----------
    Rendering SLS 'base:unbound' failed: Jinja variable 'dict object' has no attribute '10.112.24.13'
/var/cache/salt/minion/files/base/nodes/settings.sls(4):
---
# https://tedops.github.io/how-to-find-default-active-ethernet-interface.html
{% set private_address = salt['network.interface_ip'](salt['network.default_route']('inet')[0]['interface']) %}
{% set host_details = salt['pillar.get']('hosts')[private_address] %}
{% set has_public_address = host_details['public_address'] %}    <======================
{% set instance_id = host_details['instance_id'] %}
{% set instance_type = host_details['instance_type'] %}

{% set host = {} %}
{% do host.update({
[...]
---
----------
    Rendering SLS 'base:metering' failed: Jinja variable 'dict object' has no attribute '10.112.24.13'
/var/cache/salt/minion/files/base/nodes/settings.sls(4):
---
# https://tedops.github.io/how-to-find-default-active-ethernet-interface.html
{% set private_address = salt['network.interface_ip'](salt['network.default_route']('inet')[0]['interface']) %}
{% set host_details = salt['pillar.get']('hosts')[private_address] %}
{% set has_public_address = host_details['public_address'] %}    <======================
{% set instance_id = host_details['instance_id'] %}
{% set instance_type = host_details['instance_type'] %}

{% set host = {} %}
{% do host.update({
[...]
---
oszabo-de-worker2.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    Data failed to compile:
----------
    Rendering SLS 'base:unbound' failed: Jinja variable 'dict object' has no attribute '10.112.26.57'
/var/cache/salt/minion/files/base/nodes/settings.sls(4):
---
# https://tedops.github.io/how-to-find-default-active-ethernet-interface.html
{% set private_address = salt['network.interface_ip'](salt['network.default_route']('inet')[0]['interface']) %}
{% set host_details = salt['pillar.get']('hosts')[private_address] %}
{% set has_public_address = host_details['public_address'] %}    <======================
{% set instance_id = host_details['instance_id'] %}
{% set instance_type = host_details['instance_type'] %}

{% set host = {} %}
{% do host.update({
[...]
---
----------
    Rendering SLS 'base:metering' failed: Jinja variable 'dict object' has no attribute '10.112.26.57'
/var/cache/salt/minion/files/base/nodes/settings.sls(4):
---
# https://tedops.github.io/how-to-find-default-active-ethernet-interface.html
{% set private_address = salt['network.interface_ip'](salt['network.default_route']('inet')[0]['interface']) %}
{% set host_details = salt['pillar.get']('hosts')[private_address] %}
{% set has_public_address = host_details['public_address'] %}    <======================
{% set instance_id = host_details['instance_id'] %}
{% set instance_type = host_details['instance_type'] %}

{% set host = {} %}
{% do host.update({
[...]
---
oszabo-de-compute0.oszabo-l.xcu2-8y8x.wl.cloudera.site:
    Data failed to compile:
----------
    Rendering SLS 'base:unbound' failed: Jinja variable 'dict object' has no attribute '10.112.26.169'
/var/cache/salt/minion/files/base/nodes/settings.sls(4):
---
# https://tedops.github.io/how-to-find-default-active-ethernet-interface.html
{% set private_address = salt['network.interface_ip'](salt['network.default_route']('inet')[0]['interface']) %}
{% set host_details = salt['pillar.get']('hosts')[private_address] %}
{% set has_public_address = host_details['public_address'] %}    <======================
{% set instance_id = host_details['instance_id'] %}
{% set instance_type = host_details['instance_type'] %}

{% set host = {} %}
{% do host.update({
[...]
---
```

**Comment by horadla23** [comment]
I have changed the logic, hosts pillar is complete in case of targeted upscale too

**Comment by horadla23** [comment]
changes: 
- `hosts` pillar should be updated completely in case of targeted upscale too
- pillar save in case of `hosts` pillar uses different call (instead of various Set<Node> parameters) to get all nodes of related stack (cb stack or freeipa stack)
- ensure that there is no duplicated db query for stack to get all nodes and generate nodes list for pillar save as input
- minor refactors in order to make such logic possible
- improvement: cover targeted upscale feature with e2e

**Review by oleewere — APPROVED** [review]
lgtm, highstate is not failing now on DH

**Comment by lacikaaa** [comment]
@horadla23 please resolve conflicts

**Comment by horadla23** [comment]
> @horadla23 please resolve conflicts

sure, I am waiting for you to finish merging PRs to master :D

**Review by lacikaaa — CHANGES_REQUESTED** [review]

**Comment by horadla23** [comment]
need to retest

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java [hunk 1]
```diff
@@ -258,8 +261,11 @@ public void bootstrap(List<GatewayConfig> allGatewayConfigs, Set<Node> targets,
         }
     }
 
-    private void initializePillar(Set<Node> allNodes, ExitCriteriaModel exitModel, Set<String> gatewayTargetIpAddresses, SaltConnector sc) throws Exception {
-        OrchestratorBootstrap hostSave = new PillarSave(sc, gatewayTargetIpAddresses, allNodes);
+    private void saveHostsPillar(OrchestratorAware stack, ExitCriteriaModel exitModel,
+            Set<String> gatewayTargetIpAddresses, SaltConnector sc) throws Exception {
+        Set<? extends OrchestrationNode> allNodesForOrchestration = stack.getAllNodesForOrchestration();
+        Set<Node> allNodes = allNodesForOrchestration.stream().map(OrchestrationNode::getNode).collect(Collectors.toSet());
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java:267]: this stream could be a default implementation in the `OrchestratorAware` interface as  see it's used in multiple places

**Review by lacikaaa — APPROVED** [review]

**Comment by horadla23** [comment]
Screenshot from 2022-02-21 14-49-01

**Comment by horadla23** [comment]
(the only reason I re-requested review from Oliver is because re-request makes this PR visible of approved master PR list)

**Pull request merged**